src/Mvc/Mvc.RazorPages/src/Infrastructure/PageRequestDelegateFactory.cs

```diff
@@ -0,0 +1,131 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.Routing;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure
+{
+    internal class PageRequestDelegateFactory : IRequestDelegateFactory
+    {
+        private readonly PageActionInvokerCache _cache;
+        private readonly IReadOnlyList<IValueProviderFactory> _valueProviderFactories;
+        private readonly IModelMetadataProvider _modelMetadataProvider;
+        private readonly ITempDataDictionaryFactory _tempDataFactory;
+        private readonly MvcViewOptions _mvcViewOptions;
+        private readonly IPageHandlerMethodSelector _selector;
+        private readonly DiagnosticListener _diagnosticListener;
+        private readonly ILogger<PageActionInvoker> _logger;
+        private readonly IActionResultTypeMapper _mapper;
+        private readonly IActionContextAccessor _actionContextAccessor;
+        private readonly bool _enableActionInvokers;
+
+        public PageRequestDelegateFactory(
```

Comment:
Why do we need this ctor?

Reply:
IActionContextAccessor, though I can probably use a default value instead.